sqlite3/src/lib.rs

```diff
@@ -1066,22 +1072,27 @@ fn sqlite3_errstr_impl(rc: i32) -> *const ffi::c_char {
     const ANOTHER_ROW_AVAILABLE: &str = "another row available";
     const NO_MORE_ROWS_AVAILABLE: &str = "no more rows available";
 
-    match rc {
-        SQLITE_ABORT_ROLLBACK => ABORT_ROLLBACK.as_ptr() as *const ffi::c_char,
-        SQLITE_ROW => ANOTHER_ROW_AVAILABLE.as_ptr() as *const ffi::c_char,
-        SQLITE_DONE => NO_MORE_ROWS_AVAILABLE.as_ptr() as *const ffi::c_char,
+    let msg = match rc {
+        SQLITE_ABORT_ROLLBACK => ABORT_ROLLBACK,
+        SQLITE_ROW => ANOTHER_ROW_AVAILABLE,
+        SQLITE_DONE => NO_MORE_ROWS_AVAILABLE,
         _ => {
             let rc = rc & 0xff;
             if rc >= 0
                 && rc < ERROR_MESSAGES.len() as i32
                 && !ERROR_MESSAGES[rc as usize].is_empty()
             {
-                ERROR_MESSAGES[rc as usize].as_ptr() as *const ffi::c_char
+                ERROR_MESSAGES[rc as usize]
             } else {
-                UNKNOWN_ERROR.as_ptr() as *const ffi::c_char
+                UNKNOWN_ERROR
             }
         }
-    }
+    };
+
+    let cstring = CString::new(msg).expect("CString::new failed");
+    let msg_ptr = cstring.as_ptr();
+    std::mem::forget(cstring);
+    msg_ptr
```

Comment:
This leaks memory. I think SQLite has an error string in `sqlite3` that they keep reusing.